pkg/minikube/bootstrapper/bsutil/kverify/system_pods.go

```diff
@@ -112,15 +117,16 @@ func WaitForAppsRunning(cs *kubernetes.Clientset, expected []string, timeout tim
 	glog.Info("waiting for k8s-apps to be running ...")
 	start := time.Now()
 
-	checkRunning := func() (bool, error) {
-		if err := ExpectAppsRunning(cs, expected); err != nil {
-			return false, nil
+	checkRunning := func() error {
+		err := ExpectAppsRunning(cs, expected)
+		if err != nil {
+			glog.Warningf("expect apps running failed: %v", err)
 		}
-		return true, nil
+		return err
 	}
 
-	if err := wait.PollImmediate(kconst.APICallRetryInterval, timeout, checkRunning); err != nil {
-		return errors.Wrapf(err, "checking k8s-apps to be running")
+	if err := retry.Expo(checkRunning, kconst.APICallRetryInterval, timeout); err != nil {
```

Comment:
I chose `retry.Expo` because it returned the error message, which apparently was not something `wait.PollImmediate` could do. I kept it because it was doing the right thing: avoided DoS'ing an already very slow API server.

Even pinging once a second has a noticeable effect on CPU usage.

I don't actually need exponential back-off, but some sort of back-off is good for this particular case. I'd be fine if it was a 20% expansion instead of exponential.

Reply:
how about impleneting a new retry. Func that is not exponential but still backs off less aggressively?

Reply:
I would vote for a retry with 20% increment

Reply:
Done.